aiida_vasp/utils/vasp/nsw.py

```diff
@@ -0,0 +1,69 @@
+"""Utilities to help choose useful NSW values."""
+from aiida.common.utils import classproperty
+
+from aiida_vasp.utils.vasp.incar_param import AbstractIncarParam
+from aiida_vasp.utils.vasp.ibrion import IbrionFlags, Ibrion
+
+
+class Nsw(AbstractIncarParam):
+    """
+    Helper for using the NSW INCAR tag to get a value that makes sense considering related tags.
+
+    :param value: the desired number of ionic steps
+    :param ibrion: the IBRION tag value
+    :param incar_dict: a dict with normalized (lower case) incar key-value pairs
+    """
+    DEFAULT_STEP_NUM = 20
+
+    def __init__(self, value=None, ibrion=None, incar_dict=None):
```

Comment:
The intended use is:

```python
ipython > incar_d = AttributeDict()
ipython > ...
ipython > nsw = Nsw(value=<what I think it should be>, incar_dict=incar_d)
ValidationError: ... (potentially)
ipython > # or if it was fine
ipython > incar_d.nsw = nsw.value
```

Don't you think this can be useful? Do you see how it extends to scripts and complex workchains, which may have the task of modifying an unknown incar dictionary at some point?

I agree that we should rely on VASP defaults as much as we can, these classes are for when we can't.

btw: what is the default NSW value for IBRION=2? If there is one we should use that by default in the relax workchain.

Reply:
I see, I was under the impression that for position relaxation we would have to set a value anyway. In this case it is more sensible to only check in case the user wants to set a value (and log warnings).

Reply:
Yes, I guess so. Otherwise we add boiler code. Might be useful, but let us not do it before we see a use for it.